includes/class-reader-activation.php

```diff
@@ -59,7 +59,9 @@ public static function init() {
 			\add_action( 'wp_footer', [ __CLASS__, 'render_auth_form' ] );
 			\add_action( 'template_redirect', [ __CLASS__, 'process_auth_form' ] );
 			\add_filter( 'amp_native_post_form_allowed', '__return_true' );
-			\add_action( 'newspack_newsletters_add_contact', [ __CLASS__, 'register_newsletters_contact' ], 10, 2 );
+			\add_action( 'newspack_newsletters_before_add_contact', [ __CLASS__, 'register_newsletters_contact' ], 10, 2 );
+			\add_action( 'newspack_newsletters_update_contact_lists', [ __CLASS__, 'newspack_newsletters_update_contact_lists' ], 10, 5 );
+			\add_filter( 'newspack_newsletters_contact_data', [ __CLASS__, 'newspack_newsletters_contact_data' ], 10, 3 );
```

Comment:
Can we move this logic to its own class, maybe under `plugins/` directory or even `includes/class-newsletters.php`? I feel like this is going beyond the "core" reader activation functionality and the class is getting unnecessarily big.

Reply:
Done in 7077d096655d1ba3f12cb71fe92f012007e107e8